dataset/models/tf/inception_v1.py

```diff
@@ -0,0 +1,229 @@
+""" Contains inception_v1 network: https://arxiv.org/abs/1409.4842 """
+import sys
+
+import tensorflow as tf
+sys.path.append('..')
+
+from dataset.models.tf import TFModel
+from dataset.models.tf.layers import conv_block
+from dataset.models.tf.layers.pooling import max_pooling, global_average_pooling
+
+FILTERS = [
+           [[64, 96, 128, 16, 32, 32],
+           [128, 128, 192, 32, 96, 64]],
+
+           [[192, 96, 208, 16, 48, 64],
+           [160, 112, 224, 24, 64, 64],
+           [128, 128, 256, 24, 64, 64],
+           [112, 144, 288, 32, 64, 64],
+           [256, 160, 320, 32, 128, 128]],
+
+           [[256, 160, 320, 32, 128, 128],
+           [384, 192, 384, 48, 128, 128]]
+           ]
+
+NAMES = ['a', 'b', 'c', 'd', 'e']
+
+class InceptionV1(TFModel):
+    """ implementation of inception_v1 model
+
+    **Configuration**
+    -----------------
+    inputs : dict
+        dict with keys 'images' and 'labels' (see :meth:`._make_inputs`)
+    
+    dim: int {1, 2, 3}
+        spatial dimension of input without the number of channels
+
+    batch_norm : bool
+        if True enable batch normalization layers
+
+    data_format: str {'channels_last', 'channels_first'}
+
+    dropout_rate: float
+
+    head_type: str {'dense', 'conv'}
+         determine head of model
+
+         'dense' - consist of GAP and dense layers
+
+         'conv' - consist of conv 1x1 and GAP layers
+
+    """
+    def _build(self, *args, **kwargs):
+        _ = args
+
+        dim = self.get_from_config('dim', 2)
+        b_norm = self.get_from_config('batch_norm', True)
+        dropout_rate = self.get_from_config('dropout_rate', 0)
+        head_type = self.get_from_config('head_type', 'dense')
+        names = ['images', 'labels']
+        _, transformed_placeholders = self._make_inputs(names)
+
+        num_classes = self.num_classes('labels')
+        data_format = self.data_format('images')
+
+        max_pool = {'pool_size': 3,
```

Comment:
There is no point in naming args for max_pooling as `max_pool` instead of `max_pooling`

Reply:
i'm use max_pooling layer in `block`. Rename `max_pool` to `max_pooling` will make redefined-outer-name error

Reply:
You don't need `max_pool` here at all